src/format.rs

```diff
@@ -187,19 +187,57 @@ impl PayloadParams {
         let c1 = o.spec;
 
         if c0 == c1 {
+            // Exact match
             return Some(100);
-        } else {
-            // TODO: fuzzy matching.
         }
 
-        // No match
+        // Attempt fuzzy matching, since we don't have an exact match
+        if c0.codec != c1.codec || (c0.codec == Codec::Unknown && c1.codec == Codec::Unknown) {
+            // Codecs must match
+            return None;
+        }
+
+        if c0.clock_rate != c1.clock_rate {
+            // Clock rates must match
+            return None;
+        }
+
+        if c0.channels != c1.channels {
+            // Channels must match
+            return None;
+        }
+
+        if c0.codec.is_audio() {
+            let mut score: usize = 100;
+            // If neither value is specified both sides should assume the default value, 3.
+            let either_p_time_specified =
+                c0.format.min_p_time.is_some() || c1.format.min_p_time.is_some();
+            if either_p_time_specified && c0.format.min_p_time != c1.format.min_p_time {
+                score = score.saturating_sub(1);
+            }
+
+            // If neither value is specified both sides should assume FEC is not used as this is
+            // the default.
+            let either_fec_specified =
+                c0.format.use_inband_fec.is_some() || c1.format.use_inband_fec.is_some();
+
+            if either_fec_specified && c0.format.use_inband_fec != c1.format.use_inband_fec {
+                score = score.saturating_sub(10);
```

Comment:
Could use 2 here. So the sequence goes 1, 2, 4, 8 etc.

Reply:
Sure